backend/msgpack/conn.go

```diff
@@ -160,12 +155,13 @@ func (c *connection) checkReconnectWithLock() error {
 	// and we haven't past the maximum duration since the last time we attempted
 	// to connect then we simply return false without reconnecting.
 	tooManyFailures := c.numFailures > c.threshold
-	if !tooManyFailures && c.nowFn().UnixNano() < c.lastConnectNanos+int64(c.maxDuration) {
+	sufficientTimePassed := c.nowFn().UnixNano()-c.lastConnectNanos >= c.maxDuration.Nanoseconds()
+	if !tooManyFailures && !sufficientTimePassed {
 		return errNoActiveConnection
 	}
-	err := c.resetAndConnectWithLock()
+	err := c.connectWithLockFn()
```

Comment:
Won't this make us retry more often since we're not resetting the number of failures? Is that the behavior we want?

Reply:
Possibly, although in practice the time elapsed between two consecutive retries probably will be same as before when the queue is backed up since it takes almost no time to drop the buffers. Since it's currently the "known good" behavior without causing significant perf degradations on the agg tier, for now I'd like to experiment with this so we don't block all the following changes from being rolled out.